packages/java-parser/src/tokens.js

```diff
@@ -39,7 +39,25 @@ FRAGMENT("HexDigit", "[0-9a-fA-F]");
 FRAGMENT("HexDigits", "{{HexDigit}}(({{HexDigit}}|'_')*{{HexDigit}})?");
 FRAGMENT("FloatTypeSuffix", "[fFdD]");
 FRAGMENT("LineTerminator", "(\\x0A|(\\x0D(\\x0A)?))");
-
+FRAGMENT("UnicodeMarker", "uu*");
+FRAGMENT("UnicodeEscape", "\\{{UnicodeMarker}}{{HexDigit}}{4}");
+FRAGMENT("RawInputCharacter", "\\{{UnicodeMarker}}[0-9a-fA-F]{4}");
+FRAGMENT("UnicodeInputCharacter", "[{{UnicodeEscape}}{{RawInputCharacter}}]");
+FRAGMENT("OctalDigit", "[0-7]");
+FRAGMENT("ZeroToThree", "[0-3]");
+FRAGMENT(
+  "OctalEscape",
+  "\\\\({{OctalDigit}}|{{ZeroToThree}}?{{OctalDigit}}{2})"
+);
+FRAGMENT("EscapeSequence", "\\\\[btnfr\"'\\\\]|{{OctalEscape}}");
+FRAGMENT(
+  "CharacterLiteral",
+  "'(?:[^\\\\']|(?:(?:{{EscapeSequence}})|\\\\u[0-9a-fA-F]{4}))'"
+);
+FRAGMENT(
+  "StringLiteral",
+  '"(?:[^\\\\"]|(?:(?:{{EscapeSequence}})|\\\\u[0-9a-fA-F]{4}))*"'
```

Comment:
Not only does this not align with spec terminology
it also does duplication

Reply:
Could you please see the changes?